### PR TITLE
Fix Jetpack Async Specs

### PR DESCRIPTION
**Pull request by alisterscott**
Updates some page objects to be async and fixes some that broke the Jetpack e2e tests

### CODE REVIEW

File: lib/pages/wp-admin/wp-admin-sidebar.js [hunk 1]
```diff
@@ -2,14 +2,16 @@
...
-export default class WPAdminSidebar extends BaseContainer {
+export default class WPAdminSidebar extends AsyncBaseContainer {
 	constructor( driver ) {
-		driverHelper.refreshIfJNError( driver );
 		super( driver, by.css( '#adminmenumain' ) );
+	}
+
+	async _postInit() {
 		if ( driverManager.currentScreenSize() === 'mobile' ) {
 			this.driver
```
Comment (brbrr) [inline lib/pages/wp-admin/wp-admin-sidebar.js]: missing `await`?

File: lib/pages/wp-admin/wp-admin-sidebar.js [hunk 2]
```diff
@@ -22,14 +24,14 @@ export default class WPAdminSidebar extends BaseContainer {
 		}
 	}
 
-	selectPlugins() {
+	async selectPlugins() {
 		const plugInMenuSelector = by.css( '#menu-plugins' );
 		const plugInMenuItemSelector = by.css( '#menu-plugins li a[href="plugins.php"]' );
 
 		return this._selectMenuItem( plugInMenuSelector, plugInMenuItemSelector );
```
Comment (brbrr) [inline lib/pages/wp-admin/wp-admin-sidebar.js]: missing `await`?

File: lib/pages/wp-admin/wp-admin-sidebar.js [hunk 3]
```diff
@@ -38,42 +40,42 @@ export default class WPAdminSidebar extends BaseContainer {
 		return this._selectMenuItem( jetpackMenuSelector, menuItemSelector );
```
Comment (brbrr) [inline lib/pages/wp-admin/wp-admin-sidebar.js]: missing `await`?

File: lib/pages/wp-admin/wp-admin-sidebar.js [hunk 4]
```diff
@@ -38,42 +40,42 @@ export default class WPAdminSidebar extends BaseContainer {
...
 	}
 
-	selectJetpackSettings() {
+	async selectJetpackSettings() {
 		const jetpackMenuSelector = by.css( '#toplevel_page_jetpack' );
 		const menuItemSelector = by.css( '#toplevel_page_jetpack li a[href$="jetpack#/settings"]' );
 
 		return this._selectMenuItem( jetpackMenuSelector, menuItemSelector );
```
Comment (brbrr) [inline lib/pages/wp-admin/wp-admin-sidebar.js]: missing `await`?

File: lib/pages/wp-admin/wp-admin-sidebar.js [hunk 5]
```diff
@@ -38,42 +40,42 @@ export default class WPAdminSidebar extends BaseContainer {
...
 	}
 
-	selectSettingsSharing() {
+	async selectSettingsSharing() {
 		const settingsSelector = by.css( '#menu-settings' );
 		const itemSelector = by.css( '#menu-settings a[href$="sharing"]' );
 
 		return this._selectMenuItem( settingsSelector, itemSelector );
```
Comment (brbrr) [inline lib/pages/wp-admin/wp-admin-sidebar.js]: missing `await`?

File: lib/pages/wp-admin/wp-admin-sidebar.js [hunk 6]
```diff
@@ -38,42 +40,42 @@ export default class WPAdminSidebar extends BaseContainer {
...
 	}
 
-	selectSnippets() {
+	async selectSnippets() {
 		const settingsSelector = by.css( '#toplevel_page_snippets' );
 		const itemSelector = by.css( '#toplevel_page_snippets a.wp-first-item[href$="snippets"]' );
 
 		return this._selectMenuItem( settingsSelector, itemSelector );
```
Comment (brbrr) [inline lib/pages/wp-admin/wp-admin-sidebar.js]: missing `await`?

File: lib/pages/wp-admin/wp-admin-sidebar.js [hunk 7]
```diff
@@ -38,42 +40,42 @@ export default class WPAdminSidebar extends BaseContainer {
...
 	}
 
-	selectAppearanceEditCSS() {
+	async selectAppearanceEditCSS() {
 		const settingsSelector = by.css( '#menu-appearance' );
 		const itemSelector = by.css( '#menu-appearance a[href$="editcss"]' );
 
 		return this._selectMenuItem( settingsSelector, itemSelector );
```
Comment (brbrr) [inline lib/pages/wp-admin/wp-admin-sidebar.js]: missing `await`?

File: lib/pages/wp-admin/wp-admin-sidebar.js [hunk 8]
```diff
@@ -38,42 +40,42 @@ export default class WPAdminSidebar extends BaseContainer {
...
 	}
 
-	selectAddNewUser() {
+	async selectAddNewUser() {
 		const usersSelector = by.css( '#menu-users' );
 		const itemSelector = by.css( '#menu-users a[href*="user-new"]' );
 
 		return this._selectMenuItem( usersSelector, itemSelector );
```
Comment (brbrr) [inline lib/pages/wp-admin/wp-admin-sidebar.js]: missing `await`?

File: lib/pages/wp-admin/wp-admin-sidebar.js [hunk 9]
```diff
@@ -38,42 +40,42 @@ export default class WPAdminSidebar extends BaseContainer {
...
 	}
 
-	_selectMenuItem( menuSelector, menuItemSelector ) {
+	async _selectMenuItem( menuSelector, menuItemSelector ) {
 		this.driver
```
Comment (brbrr) [inline lib/pages/wp-admin/wp-admin-sidebar.js]: missing `await`?

File: lib/pages/wp-admin/wp-admin-sidebar.js [hunk 10]
```diff
@@ -90,4 +92,8 @@ export default class WPAdminSidebar extends BaseContainer {
 		}
 		return driverHelper.clickWhenClickable( this.driver, menuItemSelector );
```
Comment (brbrr) [inline lib/pages/wp-admin/wp-admin-sidebar.js]: missing `await`?

File: lib/pages/wp-admin/wp-admin-sidebar.js [hunk 11]
```diff
@@ -2,14 +2,16 @@
...
 
-import BaseContainer from '../../base-container';
 import * as driverManager from '../../driver-manager';
 import * as driverHelper from '../../driver-helper';
+import AsyncBaseContainer from '../../async-base-container';
 
-export default class WPAdminSidebar extends BaseContainer {
+export default class WPAdminSidebar extends AsyncBaseContainer {
 	constructor( driver ) {
-		driverHelper.refreshIfJNError( driver );
```
Comment (brbrr) [inline lib/pages/wp-admin/wp-admin-sidebar.js]: Could you explain, why did you decide to move `refreshIfJNError` outside of page object?

File: lib/pages/wp-admin/wp-admin-sidebar.js (hunk 1 above)
Comment (alisterscott) [reply lib/pages/wp-admin/wp-admin-sidebar.js]: Thanks, fixed!

File: lib/pages/wp-admin/wp-admin-sidebar.js (hunk 2 above)
Comment (alisterscott) [reply lib/pages/wp-admin/wp-admin-sidebar.js]: Thanks, fixed!

File: lib/pages/wp-admin/wp-admin-sidebar.js (hunk 3 above)
Comment (alisterscott) [reply lib/pages/wp-admin/wp-admin-sidebar.js]: Thanks, fixed!

File: lib/pages/wp-admin/wp-admin-sidebar.js (hunk 4 above)
Comment (alisterscott) [reply lib/pages/wp-admin/wp-admin-sidebar.js]: Thanks, fixed!

File: lib/pages/wp-admin/wp-admin-sidebar.js (hunk 5 above)
Comment (alisterscott) [reply lib/pages/wp-admin/wp-admin-sidebar.js]: Thanks, fixed!

File: lib/pages/wp-admin/wp-admin-sidebar.js (hunk 6 above)
Comment (alisterscott) [reply lib/pages/wp-admin/wp-admin-sidebar.js]: Thanks, fixed!

File: lib/pages/wp-admin/wp-admin-sidebar.js (hunk 8 above)
Comment (alisterscott) [reply lib/pages/wp-admin/wp-admin-sidebar.js]: Thanks, fixed!

File: lib/pages/wp-admin/wp-admin-sidebar.js (hunk 9 above)
Comment (alisterscott) [reply lib/pages/wp-admin/wp-admin-sidebar.js]: Thanks, fixed!

File: lib/pages/wp-admin/wp-admin-sidebar.js (hunk 11 above)
Comment (alisterscott) [reply lib/pages/wp-admin/wp-admin-sidebar.js]: I was thinking the initialisation wouldn't happen in the right order but I think it'll be ok - I'll move it back

Comment (alisterscott) [reply lib/pages/wp-admin/wp-admin-sidebar.js]: Actually, yes I know - there can't be any async calls in the constructor - so I can't do await in there

File: lib/pages/wp-admin/wp-admin-sidebar.js (hunk 10 above)
Comment (alisterscott) [reply lib/pages/wp-admin/wp-admin-sidebar.js]: Thanks, fixed!

File: lib/components/sidebar-component.js [hunk 12]
```diff
@@ -89,7 +89,9 @@ export default class SidebarComponent extends AsyncBaseContainer {
 		const searchSelector = By.css( '.site-selector input[type="search"]' );
 		const siteSelector = By.css( `.site-selector .site a[aria-label*="${ searchString }"]` );
 
-		const searchEnabled = await driverHelper.isElementPresent( this.driver, searchSelector );
+		const searchElement = await this.driver.findElement( searchSelector );
```
Comment (alisterscott) [inline lib/components/sidebar-component.js]: The element is still in the DOM, but hidden. The previous approach of using whether the element exists didn't work

**Comment by alisterscott** [comment]
@brbrr This has passed on CircleCI - I am merging now and can address any issues in subsequent PRs

**Pull request merged**